src/control/AudioControllerImpl.cpp

```diff
@@ -0,0 +1,160 @@
+#include "AudioControllerImpl.h"
+#include "Util.h"
+
+#include <i18n.h>
+#include <XojMsgBox.h>
+
+AudioControllerImpl::AudioControllerImpl(Settings* settings, Control* control) : AudioController(settings, control)
+{
+	this->settings = settings;
+	this->control = control;
+	this->audioRecorder = new AudioRecorder(settings);
+	this->audioPlayer = new AudioPlayer(control, settings);
+}
+
+AudioControllerImpl::~AudioControllerImpl()
+{
+	delete this->audioRecorder;
+	this->audioRecorder = nullptr;
+
+	delete this->audioPlayer;
+	this->audioPlayer = nullptr;
```

Comment:
remove delete's, after replacing with  unique ptrs.
remove `this->[parameter] = nullptr;` after delete in destructors since its dead code anyway and no function can call delete on `this->[parameter]` after the constructor.
If it happens, there is a programming mistake, which should not be silently catched by a set to nullptr. The software musst crash then. 

Remove empty destructor completely. (Don't even keep defaulted destructor)

Reply:
No, do NOT remove this nullpointer, this is really important, as C++ never checks if an instance is valid, you can cast every address to an instance, and only if the values are null, you get an error accessing it.
If they are not null, you e.g. get the Exception in the Audio Player, and it's not reproducible, why this happend!

Reply:
I don't agree with you neather in this case nor in general.
But this case first:
We have a destructor here, so accessing the pointer after the destructor is from the compiler point of view NOT POSSIBLE. The compiler will remove this line if optimizations are enabled since he (think to) knows, this value isn't accessed anymore.

In general:
Accessing a nullptr value is undefined behaviour as accessing a deleted pointer. 
Casting a nullptr_t to any type is completely legal and wont result in a crash. Such a cast won't even produce code.
Also assume we are not compiling with optimizations enabled and even with debug mode enabled. Then all those deleted pointers are set to some debug values like 0xDEADBEEF. Setting it to nullptr afterwards WILL HIDE ALL BUGS (like unplaned double deletion), since casting from nullptr to any type wont result in an error. Eather, it is just undefinded behaviour (compiler and platform implementation defined). So setting it to null will prevent from errors due (unwanted) double deletion or unwanted access.

The rule should be:
Is this pointer meant to be accessed after this line?
- Yes?: set it to nullptr, so we can check if the pointer is valid. Only access it if you know the pointer is valid. 
- No?  Better don't set it to nullptr, since it could hide bugs with wrongly written concurrent code or design and access mistakes.